src/xregexp.js

```diff
@@ -238,8 +238,8 @@ function getContextualTokenSeparator(match, scope, flags) {
         // No need to separate tokens if at the beginning or end of a group
         match.input[match.index - 1] === '(' ||
         match.input[match.index + match[0].length] === ')' ||
-        // No need to separate tokens if at the beginning of a non-capturing group
-        match.input.slice(match.index - 3, 3) === '(?:' ||
+        // No need to separate tokens if at the beginning of a non-capturing group or lookahead
+        match.input.slice(match.index - 3, 3).match(/\(\?[:=!]/) ||
```

Comment:
What happens if `match.index - 3` is e.g. `-1`? Even if this works, it's not precise in its meaning so might be better to clarify.

Also, this should be using `nativ.match` or preferably `nativ.test` (`nativ.test.call(regex, str)`), at least as long as those still exist. XRegExp never uses native regex methods directly since (until v4.0.0) the native methods could be overridden. As a minor optimization, the regex should start with `^` so it isn't tested at all of the string's character positions if it fails to match.

Reply:
Thanks for pointing this out, I switched to `nativ.test` in https://github.com/slevithan/xregexp/pull/196/commits/2bdd98250a7cc08e60e56a59c0b362f6657ce1ee. Also, it turns out that `slice` was the wrong method to use, so I switched to `substr` in https://github.com/slevithan/xregexp/pull/196/commits/75c8402d30692f77491b7222e4cb58b607673029, and documented its behavior with negative indices in https://github.com/slevithan/xregexp/pull/196/commits/fd9940b84888490e1b85a33afc31366bb68a69b2.